source/solvers/navier_stokes_base.cc

```diff
@@ -899,12 +901,27 @@ NavierStokesBase<dim, VectorType, DofsType>::refine_mesh_kelly()
   std::vector<bool> global_refine_flags(dim * tria.n_active_cells(), false);
   std::vector<bool> global_coarsen_flags(dim * tria.n_active_cells(), false);
 
-  bool first_variable(true);
+  bool   first_variable(true);
+  double coarsening_factor = mesh_controller.calculate_coarsening_factor(
+    this->triangulation->n_global_active_cells());
+
+  unsigned int maximal_number_of_elements =
```

Comment:
```suggestion
  const unsigned int maximal_number_of_elements =
```
If its not modified afterwards

Reply:
this is not a const